### ISSUE TITLE
[Tiled] Build options for tile source images (latest build)

### ISSUE THREAD

**Issue by Raflos10** [craftworkgames/MonoGame.Extended#526]
Since switching to the latest build of extended, I noticed that the importer for tiled map files (.tmx) automatically processes tile source images even if they aren't referenced in the .mgcb file. And even if they aren't "embedded" in the map file. The problem is that even if you add the tile images into the mgcb file, it still chooses to load the ones in the .tmx file. 
So currently it seems that there is no way to change build options (such as "generate mipmaps") for tile images.

**Comment by craftworkgames** [comment]
Okay, thanks for letting us know. I believe this change was made a in a recent PR. I haven't really had a chance to properly look into it but I'll keep it in mind as I work on the Tiled stuff.

Out of curiosity, do you have a real world use case for generating mipmaps for tiled images or was that just an example off the top of your head?

**Comment by stefanrbk** [comment]
That is a good point. This should not very difficult to add. When importing image files, it does bypass the .mgcb file and uses the default settings. The importer currently feeds the transparency color into the processor, so we can pass other options in with no problem. [See here in PR #521](https://github.com/craftworkgames/MonoGame.Extended/pull/521/files#diff-11581d3ffb9618112ab0565b60a7b499R62) The only issue I can see how we want to pass those options to the processor. We could pass it in as Properties for the tileset/image layer and filter them out in the processor.

**Comment by Raflos10** [comment]
Thanks for pointing out the location in the code. I just made a pull request with your suggestion of using tiled properties to specify the build parameters. Hopefully it works, I haven't tested it yet. 
Ultimately it would probably be better to have them specified in the content pipeline tool somehow.

**Comment by Raflos10** [comment]
Tested with the mipmaps property, seems to work.

**Comment by craftworkgames** [comment]
> We could pass it in as Properties for the tileset/image layer and filter them out in the processor.

Whoa, whoa. Hold on a second. Yes, we could do that but it doesn't really seem like the right solution to this problem. Right?

>  I just made a pull request with your suggestion of using tiled properties to specify the build parameters. Hopefully it works, I haven't tested it yet.

I appreciate the enthusiasm and I'm more than happy to keep the PR open for further discussion but I don't think this is really the change we should be making. Especially if it's not tested.

> Ultimately it would probably be better to have them specified in the content pipeline tool somehow.

Yes, I think we need to figure out how to make this work in the Pipeline tool.

**Comment by stefanrbk** [comment]
>> Ultimately it would probably be better to have them specified in the content pipeline tool somehow.
> 
> Yes, I think we need to figure out how to make this work in the Pipeline tool.
Unfortunately using `BuildAsset` to link assets together while building the project does bypass the Pipeline tool, so the processor options need to be passed from the asset which is already being processed. I'm not opposed to passing them as Properties on the map as it is uncommon to have to change these or use many of them.

**Comment by stefanrbk** [comment]
Only thing I would change is to probably remove the properties from the image once a match for a processor property is found. I don't think we need to pass around the property after the asset is loaded since it is only there to load the image correctly.

**Comment by Raflos10** [comment]
> I don't think this is really the change we should be making. Especially if it's not tested.

I agree it probably shouldn't be the permanent solution to this problem, I just put it together as a hack solution so that it's at least possible to have access to those build options for now. It's fine if you don't merge the changes, at least the code is out there in case someone else needs it. 
If I run into any other problems I'll let you guys know. Great project by the way!

**Comment by stefanrbk** [comment]
I looked at MonoGame's documentation for [ContentProcessorContext](http://www.monogame.net/documentation/?page=T_Microsoft_Xna_Framework_Content_Pipeline_ContentProcessorContext)

**Comment by Raflos10** [comment]
@stefanrbk 
Oh please, you flatter me pokéfriend.

**Comment by Raflos10** [comment]
Right... For some reason I didn't think you could use switch statements with strings. Good to know!

**Comment by craftworkgames** [comment]
> Right... For some reason I didn't think you could use switch statements with strings. Good to know!

It's part of a relatively new feature in C# called [pattern matching](https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching#using-pattern-matching-switch-statements). It can do all sorts of cool things 😄

**Comment by Raflos10** [comment]
Wow, it just happens that I was looking for a way to use switch statements with types so this helps me a lot. Thanks!

**Comment by AristurtleDev** [comment]
Closing this issue as part of the tilemap system replacement.

The legacy `TiledMap` implementation has been deprecated and is being fully replaced by the new format-agnostic tilemap architecture tracked in #1113.

No additional feature work or non-critical fixes are being made to the legacy system. The new `Tilemap` + parser + renderer architecture resolves this class of issues by design.

If you are still using `TiledMap`, please plan to migrate to the new tilemap API before the next major release, where the legacy system will be removed.

If after migrating to the new system you encounter equivalent behavior, please open a new issue referencing #1113 with a minimal repro.

Thank you.
